### ISSUE TITLE
TypeScript errors in `github/lib/github.d.ts`

### ISSUE THREAD

**Issue by RyanCavanaugh** [actions/toolkit#199]
**Describe the bug**

The shipped `github.d.ts` file contains errors. The originating code is here: https://github.com/actions/toolkit/blob/master/packages/github/src/github.ts

**To Reproduce**

1. `npm init -y`
2. `tsc --init`
3. Add `a.ts`:
```ts
import * as x from "@actions/github"
```
4. `tsc`

Result:
```
node_modules/@actions/github/lib/github.d.ts:1:49 - error TS7016: Could not find a declaration file for module '@octokit/graphql'. 'D:/Throwaway/IssuesTest/node_modules/@octokit/graphql/index.js' implicitly has an 'any' type.
  Try `npm install @types/octokit__graphql` if it exists or add a new declaration (.d.ts) file containing `declare module '@octokit/graphql';`

1 import { GraphQlQueryResponse, Variables } from '@octokit/graphql';
                                                  ~~~~~~~~~~~~~~~~~~
```

**Expected behavior**

There shouldn't be errors

**Screenshots**

**Desktop (please complete the following information):**
 - OS: I'm on Windows

**Additional context**

This is a very confusing error because the next logical step for a developer would be to `npm install @octokit/graphql`, but if you do this, it just changes the error:

```
node_modules/@actions/github/lib/github.d.ts:1:10 - error TS2305: Module '"../../../../../../Throwaway/IssuesTest/node_modules/@octokit/graphql/dist-types"' has no exported member 'GraphQlQueryResponse'.

1 import { GraphQlQueryResponse, Variables } from '@octokit/graphql';
           ~~~~~~~~~~~~~~~~~~~~

node_modules/@actions/github/lib/github.d.ts:1:32 - error TS2305: Module '"../../../../../../Throwaway/IssuesTest/node_modules/@octokit/graphql/dist-types"' has no exported member 'Variables'.

1 import { GraphQlQueryResponse, Variables } from '@octokit/graphql';
                                 ~~~~~~~~~
```
It wasn't clear to me why the `.ts` file loads the checked-in `.d.ts` instead of just pulling in the real `@octokit/grapqhl` module? This pattern is guaranteed to break for `.d.ts` emit unless some other source of type data provides a definition for the module.

Two possible fixes:
 * Update https://github.com/actions/toolkit/blob/master/packages/github/src/%40types/%40octokit/index.d.ts to better match the existing `@octokit/graphql` definition 
 * Add a dependency from `@actions/github` -> `@octokit/graphql`

Either way the import needs to be updated

**Comment by RyanCavanaugh** [comment]
Digging in a bit deeper while looking into a fix. The `.d.ts` checked in to this repo for `@octokit/graphql` describes the `2.0.1` shape (a top-level item named `defaults`); this shape no longer exists at latest (`4.3.0`). But `2.0.1` doesn't provide a `.d.ts` file, so installing that doesn't really help.

**Comment by Skycoder42** [comment]
Is there a workaround for this issue until it was fixed?

**Comment by johanneslumpe** [comment]
@Skycoder42 You can put a `d.ts` file into a local `@types` folder which contains this snippet:

```
declare module '@octokit/graphql' {
  export type Variables = any
  export type GraphQlQueryResponse = any
}
```

It seems that these are the only used exports and while this does basically disable type-checking for that part of the code, it will allow your action to properly compile.

**Comment by thboop** [comment]
@hross FYI,

**Comment by hross** [comment]
The other option is to turn off noImplicitAny

**Comment by johanneslumpe** [comment]
@hross that should probably be the last thing to try to fix something :)

**Comment by hross** [comment]
Fixed in #228
